GLES2 Can't alter projection matrix

**Godot version:**
Godot 3.1 commit 115885b1bff767a905ce98ba739fad15c5925ee0

**OS/device including version:**
Windows 10, NVidia 1060 GTX

**Issue description:**
I'm using a trick in my VR version of my racing game tutorial where I alter the project matrix to overrule the Z-output of the projection matrix. This works fine with GLES3 but with GLES2 this gives an error:

```
ERROR: _display_error_with_code: SceneShaderGLES2: Vertex shader compilation failed:
0(465) : error C7563: assignment to uniform projection_matrix
0(466) : error C7563: assignment to uniform projection_matrix
```

**Steps to reproduce:**
Use this as a vertex shader:
```
void vertex() {
	// adjust our projection to place our resulting z at our near plane
	PROJECTION_MATRIX[2][2] = 0.0;
	PROJECTION_MATRIX[3][2] = -0.1;
}
```
Then try and use it in GLES2

**Minimal reproduction project:**
Try using:
https://github.com/BastiaanOlij/vehicle-demo
And switch to GLES2 renderer

--- a/drivers/gles2/shaders/scene.glsl
+++ b/drivers/gles2/shaders/scene.glsl
@@ -424,9 +424,12 @@ void main() {
 
 #endif
 
+	mat4 local_projection_matrix = projection_matrix;
+
 	mat4 modelview = camera_inverse_matrix * world_matrix;
 	float roughness = 1.0;
 
+#define projection_matrix local_projection_matrix
 #define world_transform world_matrix
 
 	{

--- a/servers/visual/shader_types.cpp
+++ b/servers/visual/shader_types.cpp
@@ -76,7 +76,7 @@ ShaderTypes::ShaderTypes() {
 	shader_modes[VS::SHADER_SPATIAL].functions["vertex"].built_ins["CAMERA_MATRIX"] = constt(ShaderLanguage::TYPE_MAT4);
 	shader_modes[VS::SHADER_SPATIAL].functions["vertex"].built_ins["PROJECTION_MATRIX"] = ShaderLanguage::TYPE_MAT4;
 	shader_modes[VS::SHADER_SPATIAL].functions["vertex"].built_ins["MODELVIEW_MATRIX"] = ShaderLanguage::TYPE_MAT4;
-	shader_modes[VS::SHADER_SPATIAL].functions["vertex"].built_ins["INV_PROJECTION_MATRIX"] = ShaderLanguage::TYPE_MAT4;
+	shader_modes[VS::SHADER_SPATIAL].functions["vertex"].built_ins["INV_PROJECTION_MATRIX"] = constt(ShaderLanguage::TYPE_MAT4);
 	shader_modes[VS::SHADER_SPATIAL].functions["vertex"].built_ins["TIME"] = constt(ShaderLanguage::TYPE_FLOAT);
 	shader_modes[VS::SHADER_SPATIAL].functions["vertex"].built_ins["VIEWPORT_SIZE"] = constt(ShaderLanguage::TYPE_VEC2);
 	shader_modes[VS::SHADER_SPATIAL].functions["vertex"].built_ins["OUTPUT_IS_SRGB"] = constt(ShaderLanguage::TYPE_BOOL);

